src/System.CommandLine/Parsing/ParseResultVisitor.cs

```diff
@@ -34,20 +34,33 @@ internal class ParseResultVisitor : SyntaxVisitor
             _tokenizeResult = tokenizeResult;
             _rawInput = rawInput;
 
-            _unparsedTokens = new List<string>();
-            if (unparsedTokens?.Count > 0)
+            var unparsedTokensCount = unparsedTokens?.Count ?? 0;
+            _unparsedTokens = unparsedTokensCount == 0 ? new List<string>() : new List<string>(unparsedTokensCount);
```

Comment:
```suggestion
            _unparsedTokens = unparsedTokensCount == 0 ? new List<string>(0) : new List<string>(unparsedTokensCount);
```

Reply:
Interesting point.

The [parameterless constructor for `List<T>`](https://source.dot.net/#System.Private.CoreLib/List.cs,37) does exactly the same as the [constructor with initial capacity when the initial capacity is `0`](https://source.dot.net/#System.Private.CoreLib/List.cs,52) with the added (very minimal) cost of [checking for negative capacity](https://source.dot.net/#System.Private.CoreLib/List.cs,48).

But this might be an option with improved readability and maintainability:

    _unparsedTokens = new List<string>(unparsedTokensCount);